requests_toolbelt/adapters/appengine.py

```diff
@@ -44,6 +49,57 @@ def __init__(self, validate_certificate=True, *args, **kwargs):
         super(AppEngineAdapter, self).__init__(self, *args, **kwargs)
 
     def init_poolmanager(self, connections, maxsize, block=False):
-        self.poolmanager = gaecontrib.AppEngineManager(
-            validate_certificate=self._validate_certificate
-        )
+        self.poolmanager = _AEPoolManager(self._validate_certificate)
+
+
+class _AEPoolManager(object):
+    """Implements urllib3's PoolManager API."""
+
+    def __init__(self, validate_certificate=True):
+        self.validate_certificate = validate_certificate
+
+    def connection_from_url(self, url):
+        return _AEConnection(self, url)
```

Comment:
Hmmm...I could construct an actual pool of Connections inside the PoolManager, indexed by hostname, and then return the right connection based on the incoming url's host (constructing one if necessary.) And then inside the connection's urlopen, concatenate the connection's hostname with the incoming host-relative path.

Do you think that code complexity is worth it, for the gain of reusing the AppEngineConnection and AppEngineManager objects on multiple requests to the same host? (There's no actual "connection" that's getting re-used though, since it's calling urlfetch repeatedly under the hood either way.)